apps/src/templates/ProjectTemplateWorkspaceIconV2.tsx

```diff
@@ -0,0 +1,66 @@
+import classNames from 'classnames';
+import React, {useMemo} from 'react';
+
+import {ComponentPlacementDirection} from '@cdo/apps/componentLibrary/common/types';
+import {WithTooltip} from '@cdo/apps/componentLibrary/tooltip';
+import commonI18n from '@cdo/locale';
+
+import moduleStyles from './project-template-workspace-icon-v2.module.scss';
+import darkModeStyles from '@cdo/apps/lab2/styles/dark-mode.module.scss';
+
+interface ProjectTemplateWorkspaceIconV2Props {
+  tooltipPlace?: ComponentPlacementDirection;
+  darkMode?: boolean;
+}
+
+/**
+ * Modernized version of ProjectTemplateWorkspaceIcon, which uses DSCO for the Tooltip
+ * component and a FontAwesome icon for the connected level icon.
+ * This component should be used in favor of ProjectTemplateWorkspaceIcon in the workspace
+ * header.
+ * @param tooltipPlace - The placement of the tooltip relative to the icon. Uses
+ * the DSCO ComponentPlacementDirection enum.
+ * @param darkMode - boolean to indicate if the workspace is in dark mode. This is only
+ * used to style the tooltip. The icon will inherit its color from the parent component.
+ * @returns
+ */
+const ProjectTemplateWorkspaceIconV2: React.FunctionComponent<
+  ProjectTemplateWorkspaceIconV2Props
+> = ({tooltipPlace, darkMode}) => {
+  const tooltipClassName = useMemo(() => {
```

Comment:
Should the caller just pass in a `tooltipClassName` so this is decoupled from lab2 styles?

Reply:
Also just flagging this as an FYI for @levadadenys re: our theming/dark mode discussion today

Reply:
This is a good question. These styles aren't really lab2, they are the dark mode styles Mark wants for DSCO components that don't exist yet. I could move that file again to a more generic place; it's tech debt until we have themable DSCO components, so it could just stay within lab2 since that is who is using those styles now.

Reply:
Ah gotcha - yeah curious what @levadadenys thinks since it looks like theming support is in the works

Reply:
That's correct. `Tooltips` currently not supporting Light/Dark mode. In order for component to support Light/Dark mode natively we need to use `DSCO Variables`, in order to do that - we need to have Figma designs with `DSCO Variables` first.

Currently the only component that'll support light/dark mode is `SegmentedButtons` (working on it right now) - here's a PR for some context and examples of using Light/Dark mode https://github.com/code-dot-org/code-dot-org/pull/61719

As for current implementation - I think it's optimal solution. Once we'll have support of Light/Dark mode - all we'll need to do is just remove this useMemo hook and make sure we set `[data-theme]` attribute in needed place